Lake/Config/Package.lean

```diff
@@ -415,6 +428,11 @@ def oDir (self : Package) : FilePath :=
 def modToO (mod : Name) (self : Package) : FilePath :=
   Lean.modToFilePath self.oDir mod "o"
 
+/-- The path to a module's shared library within the package. -/
+def modToSharedLib (mod : Name) (self : Package) : FilePath :=
+  -- NOTE: file name MUST be unique on Windows
+  self.irDir / s!"{mod}.{sharedLibExt}"
```

Comment:
```suggestion
  Lean.modToFilePath self.libDir mod sharedLibExt
```

I think it would make more sense to to use `modToFilePath` for consistency sake (and stick it in the `libDir` as it is a library).

Reply:
See the note, we have to put the entire module name in the file *name* to make them unique. I used ir/ since they are purely internal files, they shouldn't be used from outside because they e.g. don't link to `libleanshared`.

Reply:
@Kha ah, you meant the file name part of the path for each library has to be unique. Sorry for the misunderstanding. 

However, I do still think `libDir` is more appropriate as they essentially function as the native equivalent to `olean` and would be part of theoretical binary distribution of the package library.

Reply:
Okay, the distribution argument makes sense. Note however that we can end up with `Pkg.so` and `libPkg.so` in the same directory this way, where again you'll only want to use the latter externally.

Reply:
@Kha we could use a more complex extension like `.mod.{sharedLibExt}` to further distinguish the two.